anoncreds/src/ffi/presentation.rs

```diff
@@ -65,17 +65,34 @@ pub extern "C" fn anoncreds_create_presentation(
     self_attest_values: FfiStrList,
     master_secret: ObjectHandle,
     schemas: FfiList<ObjectHandle>,
+    // TODO: this is index-matched with schemas. Is there a better solution?
+    // since we have an objecthandle list for the schemas we can not add Ids to them
+    // Otherwise they all would need ids and we would like to seperate that
+    // Also, we kind of do the same with `self_attested_names` and `self_attested_values`
+    schema_ids: FfiStrList,
```

Comment:
In AFJ (and I think the indy-sdk public api) we do it like this:

```
{
  "<schemaId>": <schemaObject>
}
```

Would that work? That way you can't really have them out of sync

Reply:
No, I am not too sure if we can work with (hash)maps over FFI easily. As with the last point, keeping the FFI layer simple types has a very high priority for me. Since they are ordered lists it will be fine, but I am not a fan of it as well :).